### ISSUE TITLE
ResidualNetworkWithBatchNorm implemented without batch norm operation

### ISSUE THREAD

**Issue by vitorhcsousa** [udlbook/udlbook#138]
On the `ResidualNetworkWithBatchNorm` in the [Notebook 11.3: Batch normalization](https://github.com/udlbook/udlbook/blob/main/Notebooks/Chap11/11_3_Batch_Normalization.ipynb), the correct implementation could be this?

```python 
class ResidualNetworkWithBatchNorm(torch.nn.Module):
  def __init__(self, input_size, output_size, hidden_size=100):
    super(ResidualNetworkWithBatchNorm, self).__init__()
    self.linear1 = nn.Linear(input_size, hidden_size)
    self.linear2 = nn.Linear(hidden_size, hidden_size)
    self.linear3 = nn.Linear(hidden_size, hidden_size)
    self.linear4 = nn.Linear(hidden_size, hidden_size)
    self.linear5 = nn.Linear(hidden_size, hidden_size)
    self.linear6 = nn.Linear(hidden_size, hidden_size)
    self.linear7 = nn.Linear(hidden_size, output_size)
    
    # Initialize batch normalization layers
    self.bn1 = nn.BatchNorm1d(hidden_size)
    self.bn2 = nn.BatchNorm1d(hidden_size)
    self.bn3 = nn.BatchNorm1d(hidden_size)
    self.bn4 = nn.BatchNorm1d(hidden_size)
    self.bn5 = nn.BatchNorm1d(hidden_size)
    self.bn6 = nn.BatchNorm1d(hidden_size)

  def count_params(self):
    return sum([p.view(-1).shape[0] for p in self.parameters()])

  def forward(self, x):
    print_variance("Input",x)
    f = self.linear1(x)
    print_variance("First preactivation",f)
    res1 = f + self.bn1(self.linear2(f.relu()))
    print_variance("After first residual connection",res1)
    res2 = res1 + self.bn2(self.linear3(res1.relu()))
    print_variance("After second residual connection",res2)
    res3 = res2 + self.bn3(self.linear4(res2.relu()))
    print_variance("After third residual connection",res3)
    res4 = res3 + self.bn4(self.linear5(res3.relu()))
    print_variance("After fourth residual connection",res4)
    res5 = res4 + self.bn5(self.linear6(res4.relu()))
    print_variance("After fifth residual connection",res5)
    return self.linear7(res5)
```

**Comment by udlbook** [comment]
I have this:
class ResidualNetworkWithBatchNorm(torch.nn.Module):
def __init__(self, input_size, output_size, hidden_size=100):
super(ResidualNetworkWithBatchNorm, self).__init__()
self.linear1 = nn.Linear(input_size, hidden_size)
self.linear2 = nn.Linear(hidden_size, hidden_size)
self.linear3 = nn.Linear(hidden_size, hidden_size)
self.linear4 = nn.Linear(hidden_size, hidden_size)
self.linear5 = nn.Linear(hidden_size, hidden_size)
self.linear6 = nn.Linear(hidden_size, hidden_size)
self.linear7 = nn.Linear(hidden_size, output_size)
self.bn1 = nn.BatchNorm1d(hidden_size)
self.bn2 = nn.BatchNorm1d(hidden_size)
self.bn3 = nn.BatchNorm1d(hidden_size)
self.bn4 = nn.BatchNorm1d(hidden_size)
self.bn5 = nn.BatchNorm1d(hidden_size)

def count_params(self):
return sum([p.view(-1).shape[0] for p in self.parameters()])

def forward(self, x):
print_variance("Input",x)
f = self.linear1(x)
print_variance("First preactivation",f)
res1 = f+ self.linear2(self.bn1(f).relu())
print_variance("After first residual connection",res1)
res2 = res1 + self.linear3(self.bn2(res1).relu())
print_variance("After second residual connection",res2)
res3 = res2 + self.linear4(self.bn3(res2).relu())
print_variance("After third residual connection",res3)
res4 = res3 + self.linear5(self.bn4(res3).relu())
print_variance("After fourth residual connection",res4)
res5 = res4 + self.linear6(self.bn5(res4).relu())
print_variance("After fifth residual connection",res5)
return self.linear7(res5)
